sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_decorator.py

```diff
@@ -182,6 +182,9 @@ def wrapper(*args, **kwargs) -> PipelineJob:
                 "tags": tags,
             }
             if _is_inside_dsl_pipeline_func():
+                # on_init/on_finalize is not supported for pipeline component
+                if job_settings.get("on_init") is not None or job_settings.get("on_finalize") is not None:
+                    raise UserErrorException("On_init/on_finalize is not supported for subgraph.")
```

Comment:
subgraph -> pipeline component. 
We don't use 'subgraph' name public.

Reply:
got it, and update this